### PR TITLE
allow remove symbol layer from registry

### PR DESCRIPTION
**Pull request by 3nids**
otherwise a plugin cannot remove an added symbol layer when it gets unloaded

### CODE REVIEW

File: src/core/symbology/qgssymbollayerregistry.cpp [hunk 1]
```diff
@@ -97,6 +97,16 @@ bool QgsSymbolLayerRegistry::addSymbolLayerType( QgsSymbolLayerAbstractMetadata
   return true;
 }
 
+bool QgsSymbolLayerRegistry::removeSymbolLayerType( QgsSymbolLayerAbstractMetadata *metadata )
+{
+  if ( !metadata || mMetadata.contains( metadata->name() ) )
```
Comment (nyalldawson) [inline src/core/symbology/qgssymbollayerregistry.cpp:102]:
```suggestion
  if ( !metadata || !mMetadata.contains( metadata->name() ) )
```

**Review by nyalldawson — COMMENTED** [review]
I think this needs a test 🤣

**Comment by 3nids** [comment]
I have been playing with this today and it works fine now.
There is no test on the registry yet as far as I can see.

**Comment by nyalldawson** [comment]
> There is no test on the registry yet as far as I can see.

Then it's a perfect time to start them... :wink:

**Comment by 3nids** [comment]
here you go

**Review by nirvn — APPROVED** [review]

**Pull request merged**